### PR TITLE
More refactoring

### PR DESCRIPTION
**Pull request by angkoonhwee**
- claimed `isDateBetween` in `Date`
- Added more javadocs
- refactored one line in `AddressBook`

### CODE REVIEW

**Review by Xiaoyunnn — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/lesson/Date.java [hunk 1]
```diff
@@ -118,8 +118,13 @@ public Date getPreviousDate(DayOfWeek dayOfWeek) {
         return prevDate;
     }
 
+    //@@author angkoonhwee
```
Comment (eeliana) [inline src/main/java/seedu/address/model/lesson/Date.java:121]: You need to end off your claim, if not the rest of the code until EOF will be considered yours

**Review by eeliana — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/lesson/Date.java (hunk 1 above)
Comment (eeliana) [reply src/main/java/seedu/address/model/lesson/Date.java:121]: image

Comment (angkoonhwee) [reply src/main/java/seedu/address/model/lesson/Date.java:121]: oh oh okayy psps

File: src/main/java/seedu/address/model/lesson/Date.java [hunk 2]
```diff
@@ -118,15 +118,21 @@ public Date getPreviousDate(DayOfWeek dayOfWeek) {
...
+     * @param end The inclusive end date.
+     * @return True if the date is between the start and end dates, inclusive.
      */
     public boolean isDateBetween(LocalDate start, LocalDate end) {
         boolean isAfterOrEqualsStart = getLocalDate().isAfter(start) || getLocalDate().isEqual(start);
         boolean isBeforeOrEqualsEnd = getLocalDate().isBefore(end) || getLocalDate().isEqual(end);
         return isAfterOrEqualsStart && isBeforeOrEqualsEnd;
     }
 
+    //@@author eeliana
```
Comment (eeliana) [inline src/main/java/seedu/address/model/lesson/Date.java:135]: Can leave empty? Just @@author will do, some of the methods are not written by me

**Review by eeliana — APPROVED** [review]

**Pull request merged**